### ISSUE TITLE
In battles, only start attack animations after text finishes scrolling

### ISSUE THREAD

**Issue by Qiangong2** [Tuxemon/Tuxemon#1208]
This is more a QoL fix than a pressing issue. When you are battling, the attack animation starts immediately. The animation should only start after the text finishes scrolling, otherwise your eyes are pulled in two different directions. It also brings Tuxemon more in line with Pokemon.

**Comment by ultidonki** [comment]
I agree, it's just quality of life instead of a pressing issue, but I've noticed this too - related is #769 about other animations happening too soon. I also think the animation stops too soon if your tuxemon is poisoned - it'll start saying 'Your tuxemon is poisoned!' or something, and then the next turn will begin while your tuxemon is taking damage, I think? 

Currently, all of this is being caused by process_action() in tuxemon/states/combat/combat.py
The self.alert (to display the message) and animation to play are both being called at the same time. 
The HP bar is hard-coded to always update after 1 second into each action. 

I made a proof-of-concept that adds up the amount of time for each animation, and delays the next one by that amount by passing the time to delay into the Task as the 'interval' parameter.
It needed some tweaking to fix the HP bars problem too - I moved check_party_hp and animate_party_status into perform_action() from out of handle_action_queue(), which makes more sense. 
However, that requires guessing how long each animation would last, so when there's a long textbox or a long animation, the following animation plays too quickly. 

Self.alert(), and animations, can both accept callback functions to call after completion. 
It might be easier to use that instead, so each animation begins when the other one ends. 
You would also need to suppress_state_change() until the animations have all finished too - it could be done with a final callback function to set _animation_is_playing to False. 
I might work on this but because there's so many different animations that will be affected by this, and need to be tested, probably not anytime soon.
